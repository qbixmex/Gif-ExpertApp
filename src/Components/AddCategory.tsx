import { useState } from "react";

type AddCategoryProps = {
  setCategory: React.Dispatch<React.SetStateAction<string[]>>;
};

const AddCategory = ({ setCategory }: AddCategoryProps) => {
  const [inputValue, setInputValue] = useState<string>("");

  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>): void => {
    event.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategory( (categories: string[]) => [ inputValue, ...categories ] );
      setInputValue("");
    }
  };

  return (
    <form onSubmit={ handleSubmit }>
      <div className="form-group my-3">
        <input
          type="text"
          className="form-control"
          value={ inputValue }
          onChange={ handleInputChange }
        />
      </div>

      <div className="d-grid">
        <button
          type="submit"
          className="btn btn-primary"
        >
          Add
        </button>
      </div>
    </form>
  );
};

export default AddCategory;
