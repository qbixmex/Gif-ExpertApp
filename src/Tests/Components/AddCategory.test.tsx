import { shallow, ShallowWrapper } from 'enzyme';
import AddCategory from '../../Components/AddCategory';

// @ts-ignore
describe("Tests on AddCategory component", () => {

  const setCategories = jest.fn();
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
  const text = "Dragon Ball";

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
  });

  test("Should match with snapshot", () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test("Should change textbox", () => {
    let input = wrapper.find("input");
    
    input.simulate("change", { target: { value: text } });
    input = wrapper.find("input");;

    expect( input.prop("value") ).toBe( text );
  });

  test("Should call setCategory and clean textbox", () => {
    wrapper.find( "input" ).simulate( "change", { target: { value: text } } );
    wrapper.find( "form" ).simulate( "submit", { preventDefault(){} } );
    expect( setCategories ).toHaveBeenCalledTimes( 1 );
    expect( setCategories ).toHaveBeenCalledWith( expect.any( Function ) );
    expect( wrapper.find( "input" ).prop( "value" ) ).toBe( "" );
  });

  test("Should not post on submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault: () => {} });
    expect( setCategories ).not.toHaveBeenCalled();
  });
});
