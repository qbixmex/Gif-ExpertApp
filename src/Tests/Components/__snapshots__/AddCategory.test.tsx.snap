// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on AddCategory component Should match with snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group my-3"
  >
    <input
      className="form-control"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="d-grid"
  >
    <button
      className="btn btn-primary"
      type="submit"
    >
      Add
    </button>
  </div>
</form>
`;
